### PR TITLE
Download tagged frames

### PR DESCRIPTION
**Pull request by yibane-yibane**

### CODE REVIEW

File: flask_app/FlaskApp.py [hunk 1]
```diff
@@ -28,15 +29,45 @@ def run(self):
...
+        if not frames_os_path:
+            Logger.logger.info(f"Can't find tagged frame for video_id: {video_id}")
+        else:
+            if not os.path.exists(local_path_to_save_frames):
+                os.makedirs(local_path_to_save_frames)
+
+            Logger.logger.info(f"Start downloading frames to local path: {local_path_to_save_frames}")
+            for index, path in enumerate(frames_os_path):
+                frame_local_path = os.path.join(local_path_to_save_frames, f"frame{index+1}")
+                await self.azure_container_handler.save_file_to_local_path(path, frame_local_path)
```
Comment (yibane-yibane) [inline flask_app/FlaskApp.py:52]: add suffix

Comment (yibane-yibane) [inline flask_app/FlaskApp.py:52]: Only read binary

File: data_base/DataModelTransactions.py [hunk 2]
```diff
@@ -49,9 +49,11 @@ def create_and_insert_metadata_and_set_metadata_id_to_frames(frames: list, image
...
     for index, image in enumerate(images):
         fov, azi, elev, tag = metadata_utils.get_metadata_arguments(image)
-        metadata = db.get_entities(data_model=data_models.Metadata,
-                                   select_section=['id'],
-                                   attributes_filters={'fov': fov, 'azimuth': azi, 'elevation': elev, 'tag': tag},
+        metadata = db.get_entities(select_section=[getattr(data_models.Metadata, 'id')],
+                                   attributes_filters={getattr(data_models.Metadata, 'fov'): fov,
+                                                       getattr(data_models.Metadata, 'azimuth'): azi,
+                                                       getattr(data_models.Metadata, 'elevation'): elev,
+                                                       getattr(data_models.Metadata, 'tag'): tag},
```
Comment (MayoG) [inline data_base/DataModelTransactions.py]: Why using `getattr` ?

File: data_base/DataModelTransactions.py [hunk 3]
```diff
@@ -63,30 +65,35 @@ def create_and_insert_metadata_and_set_metadata_id_to_frames(frames: list, image
...
+    return db.get_entities(select_section=[getattr(data_models.Frame, 'os_path')],
+                           attributes_filters={getattr(data_models.Frame, 'video_id'): video_id,
+                                               getattr(data_models.Frame, 'index'): frame_index})
+
+
+def get_tagged_frame_path_by_video_id(video_id):
+    db = SQLAlchemyHandler()
+    db.get_entities(select_section=[getattr(data_models.Frame, 'os_path')],
+                    attributes_filters={getattr(data_models.Frame, 'video_id'): video_id,
+                                        getattr(data_models.Frame, 'metadata_id'): data_models.Metadata.id,
```
Comment (MayoG) [inline data_base/DataModelTransactions.py]: why in this line you used the `getattr` and in the where you didn't ?

File: flask_app/FlaskApp.py [hunk 4]
```diff
@@ -19,6 +19,7 @@ def __init__(self):
         self.azure_container_handler = AzureBlobContainerHandler()
 
     def run(self):
+        self.app.add_url_rule('/download_tagged_frames', view_func=self.download_tagged_frames, methods=['POST'])
```
Comment (MayoG) [inline flask_app/FlaskApp.py]: Change the url to something better.

Why using the POST methods ?

File: flask_app/FlaskApp.py [hunk 5]
```diff
@@ -33,10 +34,35 @@ def run(self):
...
 
+    async def download_tagged_frames(self):
+        video_id = request.json['video_id']
+        local_path_to_save_frames = request.json['local_path_to_save_frames']
+
+        frames = DataModelTransactions.get_tagged_frame_path_by_video_id(video_id)
+
+        frames_os_path = [frame[0] for frame in frames]
+
+        if not frames_os_path:
```
Comment (MayoG) [inline flask_app/FlaskApp.py]: Use `len` instead of not

File: flask_app/FlaskApp.py [hunk 6]
```diff
@@ -33,10 +34,35 @@ def run(self):
...
+            Logger.logger.info(f"Can't find tagged frame for video_id: {video_id}")
+        else:
+            if not os.path.exists(local_path_to_save_frames):
+                os.makedirs(local_path_to_save_frames)
```
Comment (MayoG) [inline flask_app/FlaskApp.py]: See the docs of `os.makedirs` to avoid this if.

**Review by MayoG — CHANGES_REQUESTED** [review]

**Review by MayoG — APPROVED** [review]

**Pull request merged**